evolver/serial.py

```diff
@@ -69,17 +69,23 @@ def communicate(self, cmd: SerialData):
 class PySerialEmulator:
     """For testing purposes only!."""
 
+    raw_response_map = {}
+
     @classmethod
     def Serial(cls, *args, **kwargs):
         return cls(*args, **kwargs)
 
     def __init__(self, *args, **kwargs):
         self._data = None
+        self._data_raw = None
 
     def write(self, data):
+        self._data_raw = data
         self._data = EvolverSerialUART._decode_serial_data(data, suffix=EvolverSerialUART.CMD.SEND_SUFFIX.value)
 
     def readline(self):
+        if mapped_response := self.raw_response_map.get(self._data_raw):
+            return mapped_response
```

Comment:
Why shortcircuit the emulator response handling?

Reply:
not sure I get the question - the added response mapping for testing depends on this. It enables the tests added here that use `serial_mock` fixture.

Reply:
Ah, I see now, took another look and see what's going on.